napari/layers/points/_points_utils.py

```diff
@@ -4,6 +4,80 @@
 from vispy.color.colormap import Colormap
 
 
+def create_box(data):
+    """Create the axis aligned interaction box of a list of points
+
+    Parameters
+    ----------
+    data : (N, 2) array
+        Points around which the interaction box is created
+
+    Returns
+    -------
+    box : (4, 2) array
+        Vertices of the interaction box
+    """
+    min_val = data.min(axis=0)
+    max_val = data.max(axis=0)
+    tl = np.array([min_val[0], min_val[1]])
+    tr = np.array([max_val[0], min_val[1]])
+    br = np.array([max_val[0], max_val[1]])
+    bl = np.array([min_val[0], max_val[1]])
+    box = np.array([tl, tr, br, bl])
+    return box
+
+
+def points_to_squares(points, sizes):
+    """Expand points to squares defined by their size
+
+    Parameters
+    ----------
+    points : (N, 2) array
+        Points to be turned into squares
+    sizes : (N,) array
+        Size of each point
+
+    Returns
+    -------
+    rect : (4N, 2) array
+        Vertices of the expanded points
+    """
+    rect = np.concatenate(
+        [
+            points + np.sqrt(2) / 2 * np.array([sizes, sizes]).T,
+            points + np.sqrt(2) / 2 * np.array([sizes, -sizes]).T,
+            points + np.sqrt(2) / 2 * np.array([-sizes, sizes]).T,
+            points + np.sqrt(2) / 2 * np.array([-sizes, -sizes]).T,
+        ],
+        axis=0,
+    )
+    return rect
+
+
+def points_in_box(corners, points, sizes):
+    """Determine which points are in an axis aligned box defined by the corners
+
+    Parameters
+    ----------
+    points : (N, 2) array
+        Points to be checked
+    sizes : (N,) array
+        Size of each point
+
+    Returns
+    -------
+    inside : list
+        Indices of points inside the box
+    """
+    box = create_box(corners)[[0, 2]]
+    rect = points_to_squares(points, sizes)
```

Comment:
No you if any corner is inside then it is included, I find this behaviour much more visually intuitive then either must contain the center or must contain the whole shape and it becomes very clear when drawing the bounding box what will and what will not be highlighted. I have added comments to explain

Reply:
For reference, my preference is to have the centers dictate selection, because it may otherwise be impossible to select only some points if the point diameter is too large. But, this battle can wait for another time, thanks to your comment. =)

Reply:
My preference is doubly so since this way you may select a point with a circle marker without even touching the point. =\

Reply:
this i feel we could settle with some usability testing - if most people want this i'd change it for sure, but i feel it could wait till then